Code/Tools/TestImpactFramework/Runtime/Native/Code/Source/TestEngine/Native/TestImpactNativeTestEngine.cpp

```diff
@@ -204,30 +245,50 @@ namespace TestImpact
     {
         DeleteXmlArtifacts();
 
-        const auto jobInfos = m_instrumentedTestJobInfoGenerator->GenerateJobInfos(testTargets);
-
+        const auto shardedJobInfos =
+            m_shardedInstrumentedTestJobInfoGenerator->GenerateJobInfos(GenerateTestTargetAndEnumerations(testTargets));
+        
         const auto result = RunTests(
-                m_instrumentedTestRunner.get(),
-                jobInfos,
-                testTargets,
-                NativeInstrumentedTestRunnerErrorCodeChecker,
-                executionFailurePolicy,
-                testFailurePolicy,
-                targetOutputCapture,
-                testTargetTimeout,
-                globalTimeout);
-
-            if(const auto integrityErrors = GenerateIntegrityErrorString(result);
-                !integrityErrors.empty())
-            {
-                AZ_TestImpact_Eval(
-                        integrityFailurePolicy != Policy::IntegrityFailure::Abort,
-                        TestEngineException,
-                        integrityErrors);
+            m_shardedInstrumentedTestRunner.get(),
+            shardedJobInfos,
+            testTargets,
+            NativeInstrumentedTestRunnerErrorCodeChecker,
+            executionFailurePolicy,
+            testFailurePolicy,
+            targetOutputCapture,
+            testTargetTimeout,
+            globalTimeout);
+        
+        if (const auto integrityErrors = GenerateIntegrityErrorString(result); !integrityErrors.empty())
+        {
+            AZ_TestImpact_Eval(integrityFailurePolicy != Policy::IntegrityFailure::Abort, TestEngineException, integrityErrors);
+        
+            AZ_Error("InstrumentedRun", false, integrityErrors.c_str());
+        }
 
-                AZ_Error("InstrumentedRun", false, integrityErrors.c_str());
-            }
+        return result;
+    }
 
-            return result;
+    AZStd::vector<TestTargetAndEnumeration> NativeTestEngine::GenerateTestTargetAndEnumerations(
+        const AZStd::vector<const NativeTestTarget*> testTargets) const
```

Comment:
update: Looks like this should be passed by const ref (missing `&`)

Reply:
Alternatively a `AZStd::span<const NativeTestTarget*>` can be accepted by value.
The removes the needed to construct a temporary vector, when a vector does not exist at the call site.
```suggestion
        AZStd::span<const NativeTestTarget*> testTargets) const
```

Reply:
Fixed 👍